bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/util/HexUtils.java

```diff
@@ -64,4 +67,57 @@ public static String bytesToHex(byte[] bytes) {
         Arrays.setAll(bytesObjects, n -> bytes[n]);
         return bytesObjects;
     }
+
+    /**
+     * Converts an hex string (eg. format "01-23-45") into a byte array
+     *
+     * @param hexString the hex string
+     * @param delimiter a delimiter that was placed between every two bytes
+     * @return the corresponding byte array
+     */
+    public static byte[] hexToBytes(String hexString, String delimiter) {
+        // first convert to upper case to ease the rest
+        String ucHexString = hexString.toUpperCase();
+
+        Byte[] bytesObjects = Arrays.stream(ucHexString.split(delimiter)).map(s -> {
+            if (s.length() != 2) {
+                throw new IllegalArgumentException("hexString needs to have an even length: " + hexString);
+            }
+
+            return (byte) (hexCharacterToBin(s.charAt(0)) * 16 + hexCharacterToBin(s.charAt(1)));
+        }).toArray(Byte[]::new);
+
+        byte[] bytes = new byte[bytesObjects.length];
+        for (int i = 0; i < bytesObjects.length; i++) {
+            bytes[i] = bytesObjects[i];
+        }
+        return bytes;
+    }
+
+    /**
+     * Converts an hex string (in format "0123456789ABCDEF") into a byte array
+     *
+     * @param hexString the hex string
+     * @return the corresponding byte array
+     */
+    public static byte[] hexToBytes(String hexString) {
+        return hexToBytes(hexString, "(?<=\\G.{2})");
+    }
+
+    /**
+     * Convert an upper case hex character to an integer
```

Comment:
-> ... to a byte
Would be the better fit

Reply:
Assuming you agree to that change, I just did it myself, so that it can be merged right away :-)